### PR TITLE
Use XDG_CONFIG_HOME if it's available

### PR DESCRIPTION
**Pull request by Alexander-Wilms**
I'm checking that the environment variable is defined and not empty. I've tested this on a Flatpak build of Wolfendoom on Fedora 27 KDE: https://github.com/flathub/flathub/pull/245/files

There's still this discrepancy:
```
actual configPath: /home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config
Failed to create /home/awilms/.config/gzdoom/ directory:
No such file or directory
```

More info:
```
flatpak run --command=sh com.realm667.WolfenDoom_Blade_of_Agony 
sh-4.3$ stat /home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config
  File: '/home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config'
  Size: 4096            Blocks: 8          IO Block: 4096   directory
Device: fd00h/64768d    Inode: 47189384    Links: 2
Access: (0775/drwxrwxr-x)  Uid: ( 1000/  awilms)   Gid: ( 1000/  awilms)
Access: 2018-03-11 19:25:41.241220058 +0100
Modify: 2018-03-11 18:03:02.064347001 +0100
Change: 2018-03-11 18:03:02.064347001 +0100
 Birth: -
sh-4.3$ stat /home/awilms/.config/gzdoom/
stat: cannot stat '/home/awilms/.config/gzdoom/': No such file or directory
sh-4.3$ mkdir /home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config
mkdir: cannot create directory ‘/home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config’: File exists
sh-4.3$ mkdir /home/awilms/.config/gzdoom/
mkdir: cannot create directory ‘/home/awilms/.config/gzdoom/’: No such file or directory
```

Savegames work, though:
```
Game saved. (/home/awilms/.var/app/com.realm667.WolfenDoom_Blade_of_Agony/config/auto1.zds)
```

### CODE REVIEW

File: src/posix/unix/i_specialpaths.cpp [hunk 1]
```diff
@@ -110,7 +132,9 @@ FString M_GetAppDataPath(bool create)
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/" GAMENAMELOWERCASE);
+	FString path = GetEnvironmentVariable("XDG_CONFIG_HOME", FString("$HOME/.config/"));
+	path += "/zdoom/cache/";
```
Comment (Alexander-Wilms) [inline src/posix/unix/i_specialpaths.cpp]: This line needs to be removed

**Comment by coelckers** [comment]
This will need a review from a Linux user. It looks ok to me but I'd like to have a second opinion.

File: src/posix/unix/i_specialpaths.cpp [hunk 2]
```diff
@@ -203,7 +228,8 @@ FString M_GetConfigPath(bool for_reading)
 
 FString M_GetScreenshotsPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/screenshots/");
+	FString path = GetEnvironmentVariable("XDG_CONFIG_HOME", FString("$HOME/")+GAME_DIR);
+	return path + "screenshots/";
```
Comment (Alexander-Wilms) [inline src/posix/unix/i_specialpaths.cpp]: This needs a leading slash, in cae the env variable doesn't contain a trailing slash

File: src/posix/unix/i_specialpaths.cpp [hunk 3]
```diff
@@ -130,7 +154,8 @@ FString M_GetCachePath(bool create)
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/zdoom/cache");
+	FString path = GetEnvironmentVariable("XDG_CONFIG_HOME", FString("$HOME/.config/"));
+	path += "zdoom/cache/";
```
Comment (Alexander-Wilms) [inline src/posix/unix/i_specialpaths.cpp]: This needs a leading slash, in cae the env variable doesn't contain a trailing slash

**Comment by Alexander-Wilms** [comment]
I'm only printing the path for debugging purposes. Once I know why the value of configPath inexplicably changes, I'll remove it.

**Comment by alexey-lysiuk** [comment]
You need to figure out yourself what's wrong with it or ask someone else.
I have no interest in debugging Flatpak packages via console output.
Reviewing and/or testing is fine but spending hours in laggy VM debugging this is out of scope.

**Pull request closed without merging**